src/lua/compat.lua

```diff
@@ -0,0 +1,307 @@
+-- compat.lua -- internal module intended to solve compatibility problems in
+-- different parts of Tarantool. Introduced in gh-7000, see also gh-6912.
+
+local NEW = true
+local OLD = false
+
+-- luacheck: no unused
+local internals = require('internal.compat')
+
+local options_format = {
+    default  = 'string',
+    brief    = 'string',
+    obsolete = 'string',
+}
+
+-- Contains options descriptions in following format:
+-- * default  (string)
+-- * brief    (string)
+-- * obsolete ('X.Y' version or nil)
+-- * current  (boolean, true for 'new')
```

Comment:
Will not things simpler if we'll assume that `current` may contain `nil` that means 'see the default'? This way we can use `option.current or 'default'` and `option.current or option.default` that looks quite natural.

Reply:
I don't think that is much of a simplification, I prefer to have `current` to be current boolean value for it to be passed to `action`. At this point, it doesn't seem justified to refactor it.